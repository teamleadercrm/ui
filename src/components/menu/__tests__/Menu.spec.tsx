import React from 'react';
import { render } from '@testing-library/react';
import { MenuDivider, MenuItem, MenuTitle, Menu } from '..';

describe('Component - Menu', () => {
  it('renders', async () => {
    const { asFragment } = render(
      <Menu active>
        <MenuItem label="Michael Scofield" />
        <MenuItem label="Jim Halpert" />
        <MenuDivider />
        <MenuTitle>Assistants</MenuTitle>
        <MenuItem label="Dwight Schrute" />
      </Menu>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
