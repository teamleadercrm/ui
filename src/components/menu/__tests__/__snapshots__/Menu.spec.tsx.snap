// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Menu renders 1`] = `
<DocumentFragment>
  <div
    class="box menu static active"
    data-teamleader-ui="menu"
  >
    <div
      class="outline"
      style="height: 0px;"
    />
    <ul
      class="menu-inner"
    >
      <li
        class="box invisible-menu-item"
        data-teamleader-ui="menu-item"
      >
        <button
          class="box"
        />
      </li>
      <li
        class="box display-flex"
        data-teamleader-ui="menu-item"
      >
        <button
          class="box align-items-center display-flex padding-bottom-2 padding-left-3 padding-right-3 padding-top-2 text-align-left menu-item"
          style="flex: 1 1 auto;"
        >
          <div
            class="box"
            style="flex: 1 1 auto;"
          >
            <p
              class="box text text-body darkest teal"
              data-teamleader-ui="text"
            >
              Michael Scofield
            </p>
          </div>
        </button>
      </li>
      <li
        class="box display-flex"
        data-teamleader-ui="menu-item"
      >
        <button
          class="box align-items-center display-flex padding-bottom-2 padding-left-3 padding-right-3 padding-top-2 text-align-left menu-item"
          style="flex: 1 1 auto;"
        >
          <div
            class="box"
            style="flex: 1 1 auto;"
          >
            <p
              class="box text text-body darkest teal"
              data-teamleader-ui="text"
            >
              Jim Halpert
            </p>
          </div>
        </button>
      </li>
      <hr
        class="divider"
        data-teamleader-ui="menu-divider"
      />
      <div
        class="box align-items-center display-flex padding-left-3 padding-right-3 title"
        data-teamleader-ui="menu-title"
      >
        <h4
          class="box heading heading-4 darkest teal"
          data-teamleader-ui="heading"
        >
          Assistants
        </h4>
      </div>
      <li
        class="box display-flex"
        data-teamleader-ui="menu-item"
      >
        <button
          class="box align-items-center display-flex padding-bottom-2 padding-left-3 padding-right-3 padding-top-2 text-align-left menu-item"
          style="flex: 1 1 auto;"
        >
          <div
            class="box"
            style="flex: 1 1 auto;"
          >
            <p
              class="box text text-body darkest teal"
              data-teamleader-ui="text"
            >
              Dwight Schrute
            </p>
          </div>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
